Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponentUtil.cpp

```diff
@@ -158,8 +161,133 @@ namespace AZ
 
                 return result && AZ::RPI::JsonUtils::SaveObjectToFile(path, exportData);
             }
+
+            AZ::Data::AssetId GetMaterialTypeAssetIdFromMaterialAssetId(const AZ::Data::AssetId& materialAssetId)
+            {
+                if (materialAssetId.IsValid())
+                {
+                    AZ::Outcome<AZStd::vector<AZ::Data::ProductDependency>, AZStd::string> dependencyResult = AZ::Failure(AZStd::string());
+                    AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                        dependencyResult, &AZ::Data::AssetCatalogRequestBus::Events::GetAllProductDependencies, materialAssetId);
+                    if (dependencyResult)
+                    {
+                        for (const auto& dependency : dependencyResult.GetValue())
+                        {
+                            AZ::Data::AssetInfo info;
+                            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+                                info, &AZ::Data::AssetCatalogRequests::GetAssetInfoById, dependency.m_assetId);
+                            if (info.m_assetType == azrtti_typeid<AZ::RPI::MaterialTypeAsset>())
+                            {
+                                return info.m_assetId;
+                            }
```

Comment:
You might consider continuing the loop and AZ_Assert that there is exactly one material type dependency, just in case this assumption is broken someday.

Reply:
I added a comment. Since the system only supports one material type it should be safe to assume that the first one found is correct. If that ever changes everything will need to be revisited.